p2p/module.go

```diff
@@ -0,0 +1,193 @@
+package p2p
+
+import (
+	"errors"
+	"log"
+	"net"
+	"pocket/p2p/types"
+	"pocket/shared/config"
+	"pocket/shared/modules"
+	"pocket/shared/types"
+	"sync"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	MaxInbound           uint = 128
+	MaxOutbound          uint = 128
+	WireByteHeaderLength int  = 9
+	ReadBufferSize       int  = (1024 * 4)
+	WriteBufferSize      int  = (1024 * 4)
+	ReadDeadlineMs       int  = 400
+)
+
+var (
+	Protocol = "tcp"
+	Address  = "localhost:30303"
+)
+
+type P2PModule struct {
+	modules.NetworkModule
+
+	bus    modules.Bus
+	config *config.P2PConfig
+	c      *dcodec
+
+	id           uint64
+	protocol     string
+	address      string
+	externaladdr string
+
+	inbound  pipemap
+	outbound pipemap
+
+	peerlist *types.Peerlist
+
+	listener struct {
+		sync.Mutex
+		*net.TCPListener
+	}
+	listening atomic.Bool
+
+	done    chan uint
+	ready   chan uint
+	closed  chan uint
+	errored chan uint
+
+	sink     chan types.Work
+	handlers map[types.PeerEvent][]func(...interface{})
+
+	logger struct {
+		sync.RWMutex
+		print func(...interface{}) (int, error)
+	}
+
+	err struct {
+		sync.Mutex
+		error
+	}
+}
+
+var (
+	ErrNotCreated = errors.New("Module error: P2P Module not created. Trying to start the p2p module before calling create.")
```

Comment:
1. Why not group these with the other vars/constants above?
2. Why var vs const?